.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,235 @@
+          else
+            echo "Using master for RSKj"
+            CHECKOUT_REF="master"
+          fi
+          echo "CHECKOUT_REF=$CHECKOUT_REF" >> $GITHUB_ENV
+          
+      - name: Determine branch to checkout for push 
+        if: github.event_name != 'pull_request'
+        working-directory: rskj
+        env:
+          POW_REF: ${{ github.ref }}
+        run: |
+          IS_RSKJ_REF=`git ls-remote --refs $POW_REF`
+          if test -n "${IS_RSKJ_REF}"; then
+            echo "Found matching ref in RSKj"
+            CHECKOUT_REF="$POW_REF"
+          else
+            echo "Using master for RSKj"
+            CHECKOUT_REF="master"
+          fi
+          echo "CHECKOUT_REF=$CHECKOUT_REF" >> $GITHUB_ENV
+
+      - name: Check out appropriate rskj reference
+        working-directory: rskj
+        run: |
+          git switch "${{ env.CHECKOUT_REF }}"
+
+      - name: Persist RSKJ
+        uses: actions/upload-artifact@v4
+        with:
+          name: rskj
+          path: rskj
+  
+  build_federator_node:
+    runs-on: ubuntu-latest
+    container: openjdk:8-jdk
+    needs: clone_rskj_repo
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download rskj
+        uses: actions/download-artifact@v4
+        with:
+          name: rskj
+          path: rskj
+
+      - uses: actions/cache@v4
+        name: Cache Gradle
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            DONT-COMMIT-settings.gradle
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        if: steps.cache-gradle.outputs.cache-hit != 'true'
+        run: |
+          cat <<'EOF' >temp-settings.gradle
+          includeBuild('./rskj') {
+            dependencySubstitution {
+              all { DependencySubstitution dependency ->
+                if (dependency.requested instanceof ModuleComponentSelector
+                    && dependency.requested.group == 'co.rsk'
+                    && dependency.requested.module == 'rskj-core'
+                    && (dependency.requested.version.endsWith('SNAPSHOT') || dependency.requested.version.endsWith('RC'))) {
+                  def targetProject = project(":${dependency.requested.module}")
+                  if (targetProject != null) {
+                    println('---- USING LOCAL ' + dependency.requested.displayName + ' PROJECT ----')
+                    dependency.useTarget targetProject
+                  }
+                }
+              }
+            }
+          }
+          EOF
+          if [ -d ".gradle" ]; then rm -rfv .gradle; fi
+          ./configure.sh
+          ./gradlew --no-daemon --settings-file temp-settings.gradle dependencies
```

Comment:
personally, I don't like fragmentation of github workflow code, but should others support this, I will change it.

Reply:
That's the thing, this isn't supposed to be github workflow code, it's an internal thing from powpeg that requires it to build rskj. My suggestion is exactly to separate the concerns and responsibilities and let the github workflow takes care of the workflow, and let the internal ways of build an application separated from it. Because if we change this at some point, we update the script and we don't need to change the workflow.

But again, it's a personal opinion, do as you wish. It doesn't prevent the approval.